### PR TITLE
[Blog Post] Resources - Migrating Your React Native Library to the New Architecture

### PR DESCRIPTION
**Pull request by cipolleschi**
This PR introduce a Blog Post on the React Native Website related to all the work we did to support library developer for React Native in migrating their code to the New Architecture.

### CODE REVIEW

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cortinico) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:8]:
```suggestion
**tl;dr**: We are working on improving the resources supporting the React Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the New Architecture.
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cortinico) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:3]: @cipolleschi You need to add yourself to `website/blog/authors.yml`

**Review by cortinico — APPROVED** [review]
LGTM. Minor nits 👍

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md]: thanks

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:9]:
```suggestion

<!--truncate-->

```
We should add truncation marker somewhere, since the post is quite long. Not sure if it's an ideal spot, maybe we can leave also a bit of Intro.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]:
```suggestion
In the past 6 months, we’ve added a guide on adopting the New Architecture (/docs/new-architecture-intro) and an architecture deep-dive (/architecture/overview) on Fabric. We plan to expand this to include more guides and documentation around creating TurboModules, understanding CodeGen, and more. We hope to have updates to share by the 0.70 release.
```

Let's use internal links for doc pages. Those prevents the whole app rerender on navigation which lead to more seamless and quicker page loads.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
At the moment, the **New Architecture** guide covers how to migrate your app (/docs/new-architecture-app-intro) and your libraries (/docs/new-architecture-library-intro) to support the New Architecture properly.
```

Same as above.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]: It also would be nice to add links to the certain words instead of listing the whole URL in content.
Screenshot 2022-06-17 at 09 24 04

**Review by Simek — APPROVED** [review]
I have left a few small suggestion, but generally this LGTM! 

Thank you so much for adding this summary post! 🙏 
 
One more thing, please do not forget to fix the lint before the merge. At this time it complains about cases:
Screenshot 2022-06-17 at 09 24 59

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:2]:
```suggestion
title: Helping migrate React Native libraries to the New Architecture
```
this title is more aligned with the tone, also I'm not a big fan of the "Resources -" at the start, feels like it should almost be a tag instead

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:8]:
```suggestion
**tl; dr**: We are working on improving the resources supporting the React Native New Architecture. We already released a repository to help migrating your app ([RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)) and one for your libraries ([RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)). We are also revamping the [New Architecture guide](https://github.com/facebook/react-native-website/pull/3037) on the Website and we created a [GitHub Working Group](https://github.com/reactwg/react-native-new-architecture/discussions) to answer questions related to the new Architecture.
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
Currently, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
```
(nit: "at the moment" is an italian expression ;)

Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]:
```suggestion
Currently, the **New Architecture** guide covers how to migrate your app (https://reactnative.dev/docs/new-architecture-app-intro) and your libraries (https://reactnative.dev/docs/new-architecture-library-intro) to support the New Architecture properly.
```
nit: "at the moment" is mostly an italian expression ;)

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:44]:
```suggestion
```
leftover?

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:80]:
```suggestion
- **Make sure your library is listed inside the [Libraries](https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries) section**. This will help us share a status update on the migration of your library and will help us understand which struggles library authors are facing to support you better.
```
fix

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:86]:
```suggestion
### Migration Status of Popular Libraries
```

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (kelset) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: This phrase imply this blogpost will come out before 0.69.0 - @cortinico @fortmarek we should probably revise the blogpost draft to account for this blogpost existing (maybe?)

Personally I think we should avoid publishing two blogposts back-to-back, it makes it more likely that folks will miss out on one or the other; so if we want them to go out in the span of a few days, let's make sure that the other one has a link to the previous to raise awareness

Comment (fortmarek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: Thanks for the callout @kelset - it'd be great to link to this from the 0.69 blogpost. If we could merge this by Monday and then ~ Wednesday release 0.69, that'd be a reasonable schedule imho. @cipolleschi wdyt? I don't want to rush this if you think this needs more time.

**Review by fortmarek — COMMENTED** [review]
Great overview, just a couple of minor comments from my side 🙂

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:36]: Generally adding links to the headers is something that we want to avoid, since they will always have the `#` link anyway and it might be confusing for the users.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (fortmarek) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:54]: As mentioned [here](https://github.com/facebook/react-native-website/pull/3163#discussion_r900062313), we don't want links in the headers.

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:26]: Thanks for the explanation about why it is useful to use internal links!
I missed these two spots for use words for links! 😱

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:28]: Aw, thanks! :D

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:110]: Yes, sure! 

I'm apply all the comments and suggestions right now. On Monday I'll be on vacation (I had already planned to try and publish this ASAP!), but if everything is right, feel free to push the merge button!

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:14]: Thanks! I swooped the document again looking for all the `new architecture`, `new Architecture` and `New architecture` combos and now it should be consistent! 😉

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:36]: Yup, added to the `This repo` fragment in the first sentence!

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (cipolleschi) [reply website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:38]: Amazing, thank you so much!

**Review by fortmarek — APPROVED** [review]
Looks good to me 🙌

File: website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md
Comment (Simek) [inline website/blog/2022-06-16-resources-migrating-your-react-native-library-to-the-new-architecture.md:16]:
```suggestion
React Native users leverage vast number of open source libraries for building apps. For a complete and consistent ecosystem, it is necessary that these libraries migrate such that everyone can benefit from the unlocked capabilities and performance improvements of the New Architecture.
```

Case fix, on which CI is currently failing on.

**Pull request merged**
